backend/Bindings/Edna.Bindings.User/UserExtensionConfigProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Edna.Bindings.User.Attributes;
+using LtiAdvantage.NamesRoleProvisioningService;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Extensions.Logging;
+
+namespace Edna.Bindings.User
+{
+    public class UserExtensionConfigProvider : IExtensionConfigProvider
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public UserExtensionConfigProvider(IHttpClientFactory httpClientFactory)
+        {
+            _httpClientFactory = httpClientFactory;
+        }
+
+        public void Initialize(ExtensionConfigContext context)
+        {
+            var userBindingRule = context.AddBindingRule<UserAttribute>();
+
+            userBindingRule.BindToInput(GetAllUsers);
+            userBindingRule.BindToInput(attribute => new UsersClient(attribute.ServiceUrlEnvironmentVariable, _httpClientFactory));
+        }
+
+        private async Task<Models.User[]> GetAllUsers(UserAttribute userAttribute, ValueBindingContext context)
+        {
+            UsersClient usersClient = new UsersClient(userAttribute.ServiceUrlEnvironmentVariable, _httpClientFactory);
```

Comment:
Should we create a `new UserClient` everytime we make a call to `GetAllUsers` ? Or would it be better to store it as a member variable like `_httpClientFactory`?

Reply:
again, this is how it is being done in the other bindings
if we want, I can make the changes

Reply:
Sure thing. Let's keep it as is for now.